examples/extract_data/src/extract_data_login.py

```diff
@@ -70,67 +70,69 @@ async def run():
+        is_logged_in_response = client.windows.page_query(
+            session_id, window.data.window_id, prompt=IS_LOGGED_IN_PROMPT, configuration=PageQueryConfig(
+                output_schema=logged_in_schema
+            )
         )
-        parsed_response = json.loads(is_logged_in_response["data"]["modelResponse"])
+        print("Response if user is logged in")
+        parsed_response = json.loads(is_logged_in_response.data.model_response)
         if "error" in parsed_response:
-            raise ValueError(parsed_response["error"])
+            raise ValueError(parsed_response.error)
         is_user_logged_in = parsed_response["isLoggedIn"]
 
         # Prompt the user to log in if not already logged in
         if not is_user_logged_in:
-            print(f"Log into your Glassdoor account. Press Enter once done. Live view URL: {window_info['data']['liveViewUrl']}")
+            print(f"Log into your Glassdoor account. Press Enter once done. Live view URL: {window_info.data.live_view_url}")
             input()
-            print(f"To avoid logging in again, use the profileId next time: {session['profileId']}")
+            print(f"To avoid logging in again, use the profileId next time: {session.data.profile_id}")
         else:
-            print(f"User is already logged in. Live view URL: {window_info['data']['liveViewUrl']}")
+            print(f"User is already logged in. Live view URL: {window_info.data.live_view_url}")
 
         # Navigate to the target URL
         print("Navigating to target URL...")
-        await client.windows.load_url(session["id"], window_info["data"]["windowId"], {"url": TARGET_URL})
+        client.windows.load_url(session_id, window.data.window_id, url=TARGET_URL)
         print("Prompting the AI agent, waiting for a response...")
 
-        extract_data_response = await client.windows.page_query(
-            session["id"], window_info["data"]["windowId"], {
-                "prompt": EXTRACT_DATA_PROMPT,
-                "followPaginationLinks": True,
-                "configuration": {"outputSchema": EXTRACT_DATA_OUTPUT_SCHEMA},
-            }
+        extract_data_response = client.windows.page_query(
+            session_id, window.data.window_id,
+                prompt=EXTRACT_DATA_PROMPT,
+                configuration=PageQueryConfig(
+                output_schema=EXTRACT_DATA_OUTPUT_SCHEMA
+            )
         )
-        formatted_json = json.dumps(json.loads(extract_data_response["data"]["modelResponse"]), indent=2)
+        formatted_json = json.dumps(json.loads(extract_data_response.data.model_response), indent=2)
         print("Response:\n\n", formatted_json)
 
+    except Exception as e:
+        print(e)
+    
+    finally:
         # Clean up
-        await client.windows.close(session["id"], window_info["data"]["windowId"])
-        await client.sessions.terminate(session["id"])
+        client.windows.close(session_id, window.data.window_id)
```

Comment:
assuming session_id and  window are defined

Reply:
Added the check